src/libcore/intrinsics.rs

```diff
@@ -660,6 +660,18 @@ extern "rust-intrinsic" {
     /// Executes a breakpoint trap, for inspection by a debugger.
     pub fn breakpoint();
 
+    /// Returns `true` during constant evaluation and `false` otherwise.
+    ///
+    /// # Safety
+    ///
+    /// This intrinsic allows breaking referential-transparency in `const fn`
```

Comment:
Please explain referential transparency

Reply:
I've added a link to wikipedia. Should I add a brief explanation here as well ? Maybe something like: 

> The Rust compiler assumes that it is sound to replace a call to a `const fn` with the result produced by evaluating it at compile-time. If evaluating the function at run-time were to produce a different result, or have any other observable side-effects, the behavior is undefined.

Reply:
Yes that additional comment would be very helpful